### PR TITLE
Delete lot ACAS-49

### PR DESCRIPTION
**Pull request by brianbolt**
## Description
Added functionality and tests for
 - `get_lot_dependencies`
 - `delete_lot`
Refactored some code around creating back door users

## Related Issue
ACAS-49

## How Has This Been Tested?
Ran acas client tests

### CODE REVIEW

File: acasclient/acasclient.py [hunk 1]
```diff
@@ -1761,4 +1761,24 @@ def setup_items(self, item_type, items):
             raise ValueError("item_type must be one of {}".format(allowed_types))
         resp = self.session.post("{}/api/setup/{}".format(self.url, item_type), json=items)
         resp.raise_for_status()
+        return resp.json()
+
+    def get_lot_dependencies(self, lot_corp_name):
+        """Get lot dependencies for a lot by corp name
```
Comment (bffrost) [inline acasclient/acasclient.py:1767]: It's not clear what data structure this returns. Can you expand this docstring and add some info about the data structure that's expected back?

File: tests/test_acasclient.py [hunk 2]
```diff
@@ -2501,6 +2356,383 @@ def csv_to_txt(data_file_to_upload, dir):
...
+            save_meta_lot(meta_lot)
+        self.assertEqual(len(response["errors"]), 0)
+        self.assertIn("metalot", response)
+        self.assertIn("lot", response["metalot"])
+        self.assertEqual(response["metalot"]["lot"]["project"], project.code_name)
+
+        # Now create a user that is not a cmpdreg admin and does not have access to the restricted project
+        user_client = self.create_and_connect_backdoor_user(acas_user=False, acas_admin=False, creg_user=True, creg_admin=False)
+              
+        # User should NOT be able fetch the restricted lot
```
Comment (bffrost) [inline tests/test_acasclient.py:2611]: nit: is "fetch" the right word here? It seems like this code is attempting to edit / update the meta lot

**Review by bffrost — CHANGES_REQUESTED** [review]
This is fantastic work, both the thorough and easy to follow tests, and the refactoring. I'm amazed that we can now write tests like this for ACAS!

Added some minor comments and spotted what looks like a function that got clobbered by a merge. Otherwise looks really, really great.

File: tests/test_acasclient.py [hunk 3]
```diff
@@ -2470,24 +2319,30 @@ def test_049_register_large_sdf_with_error(self):
         self.assertIn('report_files', response)
         self.assertIn('Number of entries processed: 1000', response['summary'])
         self.assertIn('Number of entries with error: 1', response['summary'])
-    
-    def test_50_delete_ls_thing(self):
```
[the remark this answers is not in the document]
Comment (brianbolt) [reply tests/test_acasclient.py:2474]: Whoops, thanks. added back.

File: acasclient/acasclient.py (hunk 1 above)
Comment (brianbolt) [reply acasclient/acasclient.py:1767]: Added a returns data structure example: https://google.github.io/styleguide/pyguide.html#doc-function-raises

**Comment by brianbolt** [comment]
Re-requesting review after fixing PR comments.

**Review by bffrost — APPROVED** [review]
LGTM!

**Comment by bffrost** [comment]
Still LGTM with the linked lots change

**Pull request merged**